modules/features/discover/src/main/java/au/com/shiftyjelly/pocketcasts/discover/view/DiscoverAdapter.kt

```diff
@@ -785,7 +810,11 @@ internal class DiscoverAdapter(
                 categoriesViewHolder.binding.lblTitle.contentDescription = tittle
             }
             categoriesViewHolder.adapter.fromListId = row.listId
-            categoriesViewHolder.adapter.submitList(row.podcasts) { onRestoreInstanceState(categoriesViewHolder) }
+            categoriesViewHolder.adapter.submitList(null)
+            categoriesViewHolder.adapter.submitList(row.podcasts)
+        } else if (row is RemainingPodcastsByCategoryRow) {
+            val remainingPodcastHolder = holder as RemainingPodcastsByCategoryViewHolder
+            remainingPodcastHolder.adapter.submitList(row.podcasts)
```

Comment:
You need to submit an empty/`null` list here as well. Otherwise there will be a delay when switching between categories for remaining podcasts.

I'd suggest adding `replaceList()` method to the adapter as well.

Reply:
It's done here: [45f7eda](https://github.com/Automattic/pocket-casts-android/pull/1975/commits/45f7eda17b1b0ee29c9870a621d2add38a965ca9)